samples/17.multilingual-bot/src/main/java/com/microsoft/bot/sample/multilingual/MultiLingualBot.java

```diff
@@ -1,2 +1,173 @@
+    private static final StringBuilder WELCOME_TEXT =
+        new StringBuilder("This bot will introduce you to translation middleware. ")
+        .append("Say 'hi' to get started.");
+    private static final String ENGLISH_ENGLISH = "en";
+    private static final String ENGLISH_SPANISH = "es";
+    private static final String SPANISH_ENGLISH = "in";
+    private static final String SPANISH_SPANISH = "it";
+
+    private UserState userState;
+    private StatePropertyAccessor<String> languagePreference;
+
+    /**
+     * Creates a Multilingual bot.
+     * @param withUserState User state object.
+     */
+    public MultiLingualBot(UserState withUserState) {
+        if (withUserState != null) {
+            userState = withUserState;
+        } else {
+            throw new NullPointerException("userState");
+        }
+        languagePreference = userState.createProperty("LanguagePreference");
+    }
+
+    /**
+     * This method is executed when a user is joining to the conversation.
+     * @param membersAdded A list of all the members added to the conversation,
+     *                     as described by the conversation update activity.
+     * @param turnContext The context object for this turn.
+     * @return A task that represents the work queued to execute.
+     */
+    @Override
+    protected CompletableFuture<Void> onMembersAdded(List<ChannelAccount> membersAdded,
+                                                     TurnContext turnContext) {
+        return this.sendWelcomeMessage(turnContext);
+    }
+
+    /**
+     * This method is executed when the turnContext receives a message activity.
+     * @param turnContext The context object for this turn.
+     * @return A task that represents the work queued to execute.
+     */
+    @Override
+    protected CompletableFuture<Void> onMessageActivity(TurnContext turnContext) {
+        if (this.isLanguageChangeRequested(turnContext.getActivity().getText())) {
+            String currentLang = turnContext.getActivity().getText().toLowerCase();
+            String lang = currentLang.equals(ENGLISH_ENGLISH) || currentLang.equals(SPANISH_ENGLISH)
+                ? ENGLISH_ENGLISH : ENGLISH_SPANISH;
+
+            // If the user requested a language change through the suggested actions with values "es" or "en",
+            // simply change the user's language preference in the user state.
+            // The translation middleware will catch this setting and translate both ways to the user's
+            // selected language.
+            // If Spanish was selected by the user, the reply below will actually be shown in spanish to the user.
+            Activity reply = MessageFactory.text(String.format("Your current language code is: %1$s", lang));
+            return languagePreference.set(turnContext, lang)
+                .thenCompose(task -> turnContext.sendActivity(reply))
+                .thenCompose(task -> userState.saveChanges(turnContext, false));
```

Comment:
You're missing the [comment](https://github.com/microsoft/BotBuilder-Samples/blob/main/samples/csharp_dotnetcore/17.multilingual-bot/Bots/MultiLingualBot.cs#L61)